fix(ai-gateway): Add missing conversion samples to the on-prem doc (#6522)
Kong/developer.konghq.com: app/ai-gateway/configure-on-prem.md

## Convert {{site.ai_gateway}} 2.0 entities to self-hosted {{site.base_gateway}} config

Use `deck file ai2kong` to convert any {{site.ai_gateway}} 2.0 decK configuration into {{site.ai_gateway}} on self-hosted {{site.base_gateway}} entities.

> AI Models, AI MCP Servers, AI Agents, and AI Policies are the entities that need conversion, since each one generates {{site.base_gateway}} Services, Routes, or plugins.

Reviewer: ```suggestion

AI Model, AI MCP Server, AI Agent, and AI Policy entities need conversion, since each one generates {{site.base_gateway}} Services, Routes, or plugins.
```

Simplifying to make this line easier to read

   ```sh
   deck gateway sync kong.yaml
   ```

### Convert an AI MCP Server

> 1. Write a decK `ai.yaml` configuration file that defines an AI MCP Server in [conversion-listener mode](/ai-gateway/entities/ai-mcp-server/#server-modes), which converts a REST endpoint into an MCP tool. For example, the following AI MCP Server, `demo-mcp`, exposes a single `get-item` tool that proxies a REST `GET` request:

Reviewer: ```suggestion
1. Write a decK `ai.yaml` configuration file using the AI Gateway 2.0 entity model that defines an AI MCP Server in [conversion-listener mode](/ai-gateway/entities/ai-mcp-server/#server-modes), which converts a REST endpoint into an MCP tool. For example, the following AI MCP Server, `demo-mcp`, exposes a single `get-item` tool that proxies a REST `GET` request:
```

   ```yaml
   _format_version: "3.0"
   _info:
     select_tags:
     - 'managed_by:deck-ai'
   services:
   - host: localhost
     name: demo-mcp
     routes:
     - name: demo-mcp-route
       paths:
       - /demo-mcp
       plugins:
       - config:
           include_consumer_groups: true
           logging:
             log_audits: false
             log_payloads: false
             log_statistics: true
           mode: conversion-listener
           tools:
           - annotations:
               title: Fetch an item by id
             description: Fetch an item by id
             method: GET
             name: get-item
             parameters:
             - description: The item id
               in: path
               name: itemId
               required: true
               schema:
                 type: string
             path: /items/{itemId}
         name: ai-mcp-proxy
   ```
   {: .no-copy-code .collapsible }
1. Sync the converted config to your self-hosted {{site.base_gateway}}:
   ```sh
   deck gateway sync kong.yaml
   ```

### Convert an AI Agent

> 1. Write a decK `ai.yaml` configuration file that defines an AI Agent using the A2A protocol. For example, the following AI Agent, `demo-agent`, proxies Agent-to-Agent traffic to an upstream agent:

Reviewer: ```suggestion
1. Write a decK `ai.yaml` configuration file using the AI Gateway 2.0 entity model that defines an AI Agent using the A2A protocol. For example, the following AI Agent, `demo-agent`, proxies Agent-to-Agent traffic to an upstream agent:
```

       targets:
         - name: gpt-5.2
           provider: openai-prod
           config:
             type: openai
             temperature: 1.0
             max_tokens: 1024
   model_providers:
     - name: openai-prod
       type: openai
       config:
         auth:
           type: basic
           headers:
             - name: Authorization
               value: "{vault://ai/openai-token}"
   policies:
     - ref: ai-gw-prompt-guard
       name: ai-gw-prompt-guard
       display_name: "AI Gateway Prompt Guard"
       type: ai-prompt-guard
       enabled: true
       config:
         deny_patterns:
           - ".*(W|w)ar.*"
   ' > ai.yaml
   ```
1. Convert the {{site.ai_gateway}} entity config to {{site.base_gateway}} 3.x config:

   ```sh
   deck file ai2kong --source ai.yaml --output-file kong.yaml
   ```
   {{site.ai_gateway}} generates the same Service, Route, `ai-model-selector`, and `ai-proxy-advanced` plugins as the AI Model example, plus an `ai-prompt-guard` plugin on the same Route, scoped to the `gpt-5-2` model:

   ```yaml
   - config:
       deny_patterns:
       - .*(W|w)ar.*
     model: gpt-5-2
     name: ai-prompt-guard
     route: openai-chat
   ```

>    {: .no-copy-code .collapsible }

Reviewer: This doesn't need to be collapsible
